R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,447 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      L2322.Supplysector_Fert
+
+    # L2322.FinalEnergyKeyword_Fert: Supply sector keywords for fertilizer sector
+    A322.sector %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["FinalEnergyKeyword"]], GCAM_region_names) %>%
+      na.omit ->
+      L2322.FinalEnergyKeyword_Fert
+
+    # 2b. Subsector information
+    # L2322.SubsectorLogit_Fert: Subsector logit exponents of fertilizer sector
+    A322.subsector_logit %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorLogit"]], GCAM_region_names) ->
+      L2322.SubsectorLogit_Fert
+
+    # L2322.SubsectorShrwt_Fert and L2322.SubsectorShrwtFllt_Fert: Subsector shareweights of fertilizer sector
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
+      L2322.SubsectorShrwt_Fert
+
+    A322.subsector_shrwt %>%
+      filter(!is.na(year.fillout)) %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]], GCAM_region_names) ->
+      L2322.SubsectorShrwtFllt_Fert
+
+    # L2322.SubsectorInterp_Fert and L2322.SubsectorInterpTo_Fert: Subsector shareweight interpolation of fertilizer sector
+    A322.subsector_interp %>%
+      filter(is.na(to.value)) %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorInterp"]], GCAM_region_names) ->
+      L2322.SubsectorInterp_Fert
+
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
+      L2322.SubsectorInterpTo_Fert
+
+    # 2c. Technology information
+    # L2322.StubTech_Fert: Identification of stub technologies of fertilizer sector
+    # Note: assuming that technology list in the shareweight table includes the full set (any others would default to a 0 shareweight)
+    A322.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]], GCAM_region_names) %>%
+      rename(stub.technology = technology) ->
+      L2322.StubTech_Fert
+
+    # L2322.GlobalTechShrwt_Fert: Shareweights of global fertilizer sector technologies
+    # Notes of the workflow: The pipeline below functions as the interpolate_and_melt
+    #                        function in the old data system. The pipeline first extracts the years
+    #                        needed in interpolation then constructs the layout with proper id tags (columns)
+    #                        for interpolation years, and finally performs interpolation using 'rule=1' as in
+    #                        old data system. For more information about 'rule' please see ?approx_fun.
+    #                        Same workflow applies at several places in this chunk.
+    A322.globaltech_shrwt %>%
+      gather(year, share.weight, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.numeric(year)) ->
+      A322.globaltech_shrwt_long
+
+    df_years <- unique(A322.globaltech_shrwt_long$year)
+    int_years <- setdiff(c(BASE_YEARS, FUTURE_YEARS), df_years)
```

Comment:
Do we need to do this? The `Name that Function` part of the wiki has code snippets to replicate this behavior: https://github.com/JGCRI/gcamdata/wiki/Name-That-Function

Reply:
Thanks! That was the old ugly way I did before the wiki got updated with the elegant solution.